### PR TITLE
RFC: Pair to Pair conversions

### PR DESCRIPTION
**Pull request by ghost**
Stumbled upon this when trying to convert `Pair{SubString,SubString}` into `Pair{String,String}`.

### CODE REVIEW

File: test/operators.jl [hunk 1]
```diff
@@ -21,6 +21,7 @@ B = [true true false]
 @test reverse(Pair(1,2)) == Pair(2,1)
 @test reverse(Pair("13","24")) == Pair("24","13")
 @test typeof(reverse(Pair{String,Int64}("a",1))) == Pair{Int64,String}
+@test convert(Pair{Float64,Float64}, 17 => 4711) == (17 => 4711)
```
Comment (tkelman) [inline test/operators.jl]: this should test the element types too

Comment (eschnett) [inline test/operators.jl]: Using `===` instead of `==` will do this.

Comment (ninjin) [inline test/operators.jl]: Got it, new version pushed.

**Review by vchuravy — APPROVED** [review]

File: base/operators.jl [hunk 2]
```diff
@@ -964,6 +964,11 @@ reverse{A,B}(p::Pair{A,B}) = Pair{B,A}(p.second, p.first)
 endof(p::Pair) = 2
 length(p::Pair) = 2
 
+convert{A,B}(::Type{Pair{A,B}}, x::Pair{A,B}) = x
+function convert{A,B,C,D}(::Type{Pair{A,B}}, x::Pair{C,D})
```
Comment (stevengj) [inline base/operators.jl]: It should be sufficient to declare `function convert{A,B}(::Type{Pair{A,B}}, x::Pair)` since you don't use `C` or `D`

Comment (ninjin) [inline base/operators.jl]: Fixed, I also added a few additional test to test symmetry between the first and second element.

**Pull request merged**